Mac build and docs improvements

vadz opened the pull request:
Trying to explain to somebody how to build wxWidgets for use with their application, I realized that our documentation, and even our `configure`, is not optimal, to put it mildly.

I still struggle to explain to people how to use wxWidgets from Xcode, so I didn't document this here, but if anybody can provide some good instructions for this, it would be great.

vadz committed:
Default to arm64 + x86_64 universal binaries for wxOSX

Using i386 by default isn't useful any more -- but using arm64 is.

vadz committed:
Recommend using parallel make when building wxWidgets

This should help at least some people spend less time waiting.

vadz committed:
Recommend building just "minimal" sample

Don't tell people to build all the samples and demos because this takes
time and some of them are badly outdated. Do tell them to build and run
the minimal one to check that things work as intended.

Also remove links to the wiki from wxOSX instructions, there is not much
useful from non-archaeological point of view there.

vadz committed:
Add section about building release Mac libraries

This doesn't seem to be documented anywhere else, but is rather
important to know.

lanurmi commented on docs/osx/install.md:
```diff
@@ -9,39 +9,63 @@ Most OS X developers should start by downloading and installing Xcode
 
-    cd samples; make;cd ..
-    cd demos;   make;cd ..
+    make -j16
```
Because `-j16` requires quite a high-end Mac system to be useful, and some people are likely to just copy-paste `-j16` without thinking how many cores or how much memory they have, I propose using a smaller figure in this example. Such as 6 or 8.

vadz replied:
OK, done, and added an extra warning about this, thanks.

vadz committed:
Reduce the degree of make parallelism in examples

And add a warning to not do this if it's not supported.

vadz commented:
@csomor just a ping to ask you if you could please check that [the change to configure](https://github.com/wxWidgets/wxWidgets/pull/24128/files#diff-49473dca262eeab3b4a43002adb08b4db31020d190caaad1594b47f1d5daa810) is correct. TIA!

vadz committed:
Use symbolic constant for macOS 10.12 version

No real changes, just use a constant instead of a hardcoded number in a
version check.

csomor reviewed (commented):
Hi @vadz, thanks for this change, I don't know how many people just building universal without indicating archs ... since only from macOS 11 arm64 can be built ... should we make this reflecting the OS it's run on to avoid error messages ?

vadz commented:
> Hi @vadz, thanks for this change, I don't know how many people just building universal without indicating archs ... since only from macOS 11 arm64 can be built ... should we make this reflecting the OS it's run on to avoid error messages ?

Well, this is the question... I think people using master are very very likely to target arm64, but I can't be sure, of course. If you think it's worth making this conditional on the OS version, I'd need to understand how to test for it (using `sw_vers -productVersion`, I guess, but what are the formats used for different versions? or maybe `uname -r`?) and what should be used for which version.

I definitely think we should remove `i386` because it's not even supported any longer.

csomor commented:
> > Hi @vadz, thanks for this change, I don't know how many people just building universal without indicating archs ... since only from macOS 11 arm64 can be built ... should we make this reflecting the OS it's run on to avoid error messages ?
> 
> Well, this is the question... I think people using master are very very likely to target arm64, but I can't be sure, of course. If you think it's worth making this conditional on the OS version, I'd need to understand how to test for it (using `sw_vers -productVersion`, I guess, but what are the formats used for different versions? or maybe `uname -r`?) and what should be used for which version.

the first segment of `sw_vers -productionVersion` returning 10 would be true for any system before macOS 11 Big Sur.
Alternatively we could ask for `gcc --version` and treat everything before Xcode 12.2 as not supporting arm64 (this is a little bit more precise as Catalina supports running Xcode up to 12.4, which can produce a universal binary 2.

[Xcode versions](https://trac.macports.org/wiki/XcodeVersionInfo)

But if you think this is more effort than it deserves, we just put a note somewhere we can link to, in case questions arise ..

> I definitely think we should remove `i386` because it's not even supported any longer.

yes

vadz committed:
Don't use arm64 for universal binaries by default under macOS 10

This architecture is supported since macOS 11.

vadz commented:
Thanks, I've added a test for the OS version as this was simpler than checking gcc version.

vadz commented:
I'm going to merge this, finally. Please let me know if I broke/misdocumented anything.
